src/main/java/com/snowflake/kafka/connector/records/RecordService.java

```diff
@@ -297,7 +316,7 @@ public SnowflakeTableRow(SnowflakeRecordContent content, JsonNode metadata) {
   }
 
   void putKey(SinkRecord record, ObjectNode meta) {
-    if (record.key() == null) {
+    if (record.key() == null || record.keySchema() == null) {
```

Comment:
why we can't put the key when the keySchema is null but not the key? This change might be risky and introduce a behavior change for both Snowpipe and Snowpipe Streaming

Reply:
a null keySchema will NPE on the next if statement, however we should never get into this situation due to previous handleNativeRecord() calls. I'll update the logic here to both put the key and handle the NPE

Reply:
the current code will NPE on the next if statement if keySchema is null. 

handleNativeRecord() prevents null keySchemas from being passed into putKey, however we should throw a better error if we ever get a null keySchema, so I added a null check.

This shouldn't be risky because we are changing the error thrown from NPE to a Snowflake error.